plasmapy/plasma/fields.py

```diff
@@ -0,0 +1,326 @@
+"""
+Defines the PosGrid and AbstractField classes and provides subclasses
+representing different field configurations
+"""
+
+__all__ = [
+    "AbstractField",
+    "NoFields",
+    "ElectrostaticGaussianSphere",
+    "AxiallyMagnetizedCylinder",
+    "ElectrostaticPlanarShock",
+    "example_fields",
+]
+
+import astropy.units as u
+import numpy as np
+
+from abc import ABC, abstractmethod
+from scipy.special import erf as erf
+
+from plasmapy.plasma import grids as grids
+
+
+class AbstractField(ABC):
+    """
+    Base class for field grids.
+    """
+
+    def __init__(self, grid: grids.AbstractGrid, emax=0 * u.V / u.m, bmax=0 * u.T):
+        r"""
+        Initialize a field object
+
+        Parameters
+        ----------
+        grid : `plasmapy.plasma.grids.AbstractGrid` or child class thereof
+            Positions of grid points in space.
+
+        emax : `~astropy.units.Quantity`
+            Maximum electric field to normalize example field to. The default
+            is 0 * u.V / u.m.
+
+        bmax : `~astropy.units.Quantity`
+            Maximum magnetic field to normalize the example field to. The
+            default is 0 * u.T.
```

Comment:
Suppose I didn't understand what this did and passed in negative values, here? Would it break?

Reply:
I think it would just give you back -B or -E: the routine just normalizes the array to the maximum value then multiplies by emax and bmax respectively.